File: acmp.ru/[0401] - [0500]/[0457] Постоянная Капрекара.py
n = int(input())
cnt = 0

while n != 6174:
    right = ''.join(sorted(str(n).ljust(4, '0')))
    left = right[::-1]
    n = int(left) - int(right)

    cnt += 1

print(n)
print(cnt)

File: acmp.ru/[0401] - [0500]/[0457] Постоянная Капрекара.cpp
#include <iostream>
#include <cstring>
#include <algorithm>
using namespace std;

int main() {
    int n, cnt = 0;
    cin >> n;

    string left, right;
    while (n != 6174) {
        left = to_string(n);
        left += string(4 - left.length(), '0');
        sort(left.begin(), left.end());

        right = left;
        reverse(left.begin(), left.end());

        n = stoi(left) - stoi(right);
        cnt++;
    }

    cout << n << endl << cnt << endl;

    return 0;
}